src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -527,7 +536,7 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
     void MainPage::_InitializeProfilesList()
     {
-        const auto menuItems = SettingsNav().MenuItems();
+        const auto menuItems = SettingsNav().MenuItemsSource().try_as<Windows::Foundation::Collections::IVector<IInspectable>>();

```

Comment:
This introduces an ordering dependency, right? It seems like there is code that _can_ run before `MenuItemsSource` is set and code that can only run after. Are we clear on what happens here if somehow it's not been set?